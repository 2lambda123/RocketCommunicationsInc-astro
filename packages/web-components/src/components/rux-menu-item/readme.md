# rux-menu-item



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description                                                                                    | Type      | Default     |
| ---------- | ---------- | ---------------------------------------------------------------------------------------------- | --------- | ----------- |
| `disabled` | `disabled` | sets the menu item as disabled                                                                 | `boolean` | `false`     |
| `selected` | `selected` | sets the menu item as selected                                                                 | `boolean` | `false`     |
| `value`    | `value`    | the value returned when item is selected. If no value is given, the text content will be used. | `any`     | `undefined` |


## Events

| Event                 | Description                                                     | Type                  |
| --------------------- | --------------------------------------------------------------- | --------------------- |
| `ruxmenuitemselected` | When a rux-menu item is selected, emits the value of that item. | `CustomEvent<object>` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
