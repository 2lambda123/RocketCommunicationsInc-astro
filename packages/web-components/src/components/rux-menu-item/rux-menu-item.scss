:host {
    display: block;
}

.rux-menu-item {
    position: relative;
    padding: 6px 8px;
    color: var(--color-interactive-default);
    cursor: pointer;
    &:hover {
        color: var(--color-interactive-hover);
        background: var(--color-background-base-hover);
    }
    &--selected {
        background: var(--color-background-surface-selected);
        color: var(--color-text-primary);
    }
    &--disabled {
        opacity: 0.4; // change to token
        cursor: not-allowed;
    }
}
