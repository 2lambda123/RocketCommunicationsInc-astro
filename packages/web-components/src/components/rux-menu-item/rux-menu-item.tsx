import {
    Prop,
    Component,
    Host,
    h,
    Watch,
    Event,
    EventEmitter,
    Listen,
    Element,
} from '@stencil/core'

@Component({
    tag: 'rux-menu-item',
    styleUrl: 'rux-menu-item.scss',
    shadow: true,
})
export class RuxMenuItem {
    @Element() el!: HTMLRuxMenuItemElement
    /**
     * @prop selected - sets the menu item as selected
     */
    @Prop({ reflect: true }) selected = false
    /**
     * @prop disabled - sets the menu item as disabled
     */
    @Prop({ reflect: true }) disabled = false
    /**
     * @prop value - the value returned when item is selected. If no value is given, the text content will be used.
     */
    @Prop() value: any

    /**
     * @event ruxmenuitemselected - When a rux-menu item is selected, emits the value of that item.
     */
    @Event({ eventName: 'ruxmenuitemselected', bubbles: true, composed: true })
    ruxMenuItemSelected!: EventEmitter<object>

    @Listen('click', { passive: true })
    handleClick() {
        if (!this.disabled) {
            const emittedValue = this.value ? this.value : this.el.textContent
            this.ruxMenuItemSelected.emit({ value: emittedValue })
        }
    }

    render() {
        return (
            <Host>
                <div
                    class={{
                        'rux-menu-item': true,
                        'rux-menu-item--selected': this.selected,
                        'rux-menu-item--disabled': this.disabled,
                    }}
                >
                    <slot></slot>
                </div>
            </Host>
        )
    }
}
