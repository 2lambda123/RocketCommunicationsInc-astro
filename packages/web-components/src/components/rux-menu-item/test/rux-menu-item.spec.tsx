import { newSpecPage } from '@stencil/core/testing'
import { RuxMenuItem } from '../rux-menu-item'

describe('rux-menu-item', () => {
    it('renders', async () => {
        const page = await newSpecPage({
            components: [RuxMenuItem],
            html: `<rux-menu-item></rux-menu-item>`,
        })
        expect(page.root).toMatchSnapshot()
    })
    it('changes to anchor tag based on an href prop', async () => {
        const page = await newSpecPage({
            components: [RuxMenuItem],
            html: `<rux-menu-item href="https://www.astrouxds.com"></rux-menu-item>`,
        })
        expect(page.root).toMatchSnapshot()
    })
})
