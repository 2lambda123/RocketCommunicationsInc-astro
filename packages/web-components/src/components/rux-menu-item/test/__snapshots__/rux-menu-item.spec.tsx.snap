// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rux-menu-item renders 1`] = `
<rux-menu-item>
  <mock:shadow-root>
    <div class="rux-menu-item">
      <slot></slot>
    </div>
  </mock:shadow-root>
</rux-menu-item>
`;
